openff/toolkit/utils/_cif_to_substructure_dict.py

```diff
@@ -165,15 +166,22 @@ def _generate_atom_symbol_number_dictionary(cif_entry):
         symbol_to_num_dict : dict
             Atom symbol to atomic number (key, value) dictionary.
         """
-        atom_symbol_list = cif_entry["_chem_comp_atom.type_symbol"]
-        elements = element(atom_symbol_list)
-        atomic_nums_list = []
-        for element_ in elements:
-            atomic_nums_list.append(element_.atomic_number)
-        # Create dictionary
-        symbol_to_num_dict = {
-            symbol: number for symbol, number in zip(atom_symbol_list, atomic_nums_list)
+        # Is this de-duplicated? i.e. can it be ["C", "C", "H"] or only ["C", "H"] ?
+        atom_symbol_list: List[str] = cif_entry["_chem_comp_atom.type_symbol"]
+
+        # Is this dict a de-duplicated mapping between element symbols and atomic numbers
+        # for the elements in this cif file? If so, can maybe do it in one step
+        """
```

Comment:
(not blocking) In the interest of time I'm not going to give feedback on this. I forgot exactly what this function expects and we don't have tests for this file yet. This is not expected to be run by users and so runtime isn't super critical as long as it's under an hour. If you're interested in optimizing/refactoring this, feel free to do so at your discretion. The best test I can suggest would be ensuring that the `openff/toolkit/utils/make_*_substructures.py` scripts yield the same outputs.

Reply:
I'll not tinker with this more, then. Except maybe to suggest it shouldn't be in the API, and instead tucked away in another older?

Reply:
I was under the impression that the underscore prepended to the file name communicated the privateness/here-be-dragons-ness of the contents. Though I'm not sure if that's a python convention in reality.

Reply:
As far as I remember when we first planned this, we didn't want to support a public infrastructure to do this, only just what's "enough" for the toolkit. I think I'm still in favor of this stance, since we know how this problem can be pretty complex. Also, I guess it's worth noticing that this `.py` file is expected to be used only once in a while. My 2 cents :). BTW, I'm loving how this is evolving, thanks for the good work!

Reply:
> Though I'm not sure if that's a python convention in reality.

It is :) https://www.python.org/dev/peps/pep-0008/?#public-and-internal-interfaces

Reply:
I'd like to clarify a bit the distinction between two different things. I'm not sure the right jargon (happy to be told what the terminology is, I can't think of anything but "private API" and "scripts") so I'll resort to being descriptive:

1. Code that's for core toolkit functionality and might get called at runtime while interacting with the public API, but is either too dangerous to be explicitly exposed to users or meant to hide away implementation details. Stuff like `Molecule._add_trivalent_lone_pair_virtual_site` or `OpenEyeToolkitWrapper._find_smarts_matches`. These code paths have intrinsic value (in the sense that they **are** the product), **should** be covered by unit tests, will be included in conda tarballs, and fall under the umbrella of code we're responsible for maintaining/documenting/etc.
2. Code that's  for one-off data cleaning or developer-facing operations, but not a part of a core feature and will not be called by a user (directly or indirectly) at runtime. These files are valuable  in large part for what they produce (i.e.  big CIF file I don't understand the details of), but aren't covered by unit tests, may or may not be included in conda tarballs, and don't have the same maintenance burden.

What I'm getting at here is that code that cannot be accessed at runtime should be moved `/utilities/` or somewhere that would not be packaged as part of the Python module. That folder is from before my time, but it looks like those files didn't contribute as much to maintenance burden.

There's no danger to the user as is since it's definitely not part of the public API. But I think functions that should be used once in a while should not be in the core module (though no less important, of course).